commands/joke.js

```diff
@@ -5,23 +5,71 @@ module.exports = {
   name: 'joke',
   description: 'Random Joke',
   async execute(msg, args) {
-    fetch('https://us-central1-dadsofunny.cloudfunctions.net/DadJokes/random/jokes', { 
-      method: 'GET'})
-      .then(response => { return response.json(); })
-      .then(json => {
-        joke = new Discord.MessageEmbed()
-          .setAuthor("Joke Bot")
-          .setColor("#c5f542")
-          .setTitle(json.setup)
-	  .setThumbnail("https://repository-images.githubusercontent.com/193169245/5462aa00-f356-11e9-846d-551973c550a6")
-      
-        msg.channel.send(joke).then(j => { 
-          joke.setDescription(json.punchline); 
-	  joke.setFooter("Joke provided by https://dadsofunny.firebaseapp.com","https://repository-images.githubusercontent.com/193169245/5462aa00-f356-11e9-846d-551973c550a6");
-          setTimeout(function () {j.edit(joke)}, 7000); 
-        });
-    
-                    
-    });
+
+    function sendDadJoke(){
+      fetch('https://us-central1-dadsofunny.cloudfunctions.net/DadJokes/random/jokes', { 
+        method: 'GET'})
+        .then(response => { return response.json(); })
+        .then(json => {
+          const joke = new Discord.MessageEmbed()
+            .setAuthor(msg.author.username+" wants jokes") 
+            .setColor("#c5f542")
+            .setTitle(json.setup)
+      .setThumbnail("https://repository-images.githubusercontent.com/193169245/5462aa00-f356-11e9-846d-551973c550a6")
+        
+          msg.channel.send(joke).then(j => { 
+            joke.setDescription(json.punchline); 
+      joke.setFooter("Joke provided by https://dadsofunny.firebaseapp.com","https://repository-images.githubusercontent.com/193169245/5462aa00-f356-11e9-846d-551973c550a6");
+            setTimeout(function () {j.edit(joke)}, 7000); 
+          });            
+      });
+    }
+
+    function sendJokeApiJoke(category){
+      console.info("Category: " + category);
+      fetch('https://v2.jokeapi.dev/joke/'+category+'?format=xml?type=twopart?blacklistFlags=racist,nsfw,explicit', { 
+        method: 'GET'})
+        .then(response => { return response.json(); })
+        .then(json => {
+          console.info(json);
+          if (json.error === false) {
+              const joke = new Discord.MessageEmbed()
+              .setAuthor(msg.author.username+" wants jokes")
+              .setColor("#c5f542")
+              .setTitle(json.setup)
+              .setThumbnail("https://sv443.net/resources/images/jokeapi.webp")
+              msg.channel.send(joke).then(j => { 
+              joke.setDescription(json.delivery); 
+              joke.setFooter("Joke provided by https://sv443.net/jokeapi/","https://sv443.net/");
```

Comment:
Second argument to setFooter is a link to an image. It will display the small image to the left of the footer. More than likely the same link in setThumbnail.

Reply:
fixed
![image](https://user-images.githubusercontent.com/1059333/114795452-6c489d80-9d43-11eb-87b9-abe53c9bf9a2.png)